paint-house.java

```diff
@@ -0,0 +1,24 @@
+// Time Complexity : O(n*m)
+// Space Complexity : O(n*m)
```

Comment:
The modifications to the cost matrix are in-place and so the Space complexity would be O(1)